internal/datasources/http/data_source_http.go

```diff
@@ -39,12 +39,26 @@ the server certificate's chain of trust. Data retrieved from servers not under
 your control should be treated as untrustworthy.`,
 
 		Attributes: map[string]tfsdk.Attribute{
+			"id": {
+				Description: "The ID of this resource.",
+				Type:        types.StringType,
+				Computed:    true,
+			},
+
 			"url": {
 				Description: "The URL for the request. Supported schemes are `http` and `https`.",
 				Type:        types.StringType,
 				Required:    true,
 			},
 
+			"method": {
+				Description: "The HTTP Method for the request. " +
+					"Allowed methods are a subset of methods defined in [RFC7231](https://datatracker.ietf.org/doc/html/rfc7231#section-4.3) namely, " +
+					"`GET`, `POST`, `PUT`, `PATCH` and `DELETE`.",
```

Comment:
Should we also allow `HEAD` here?

Yes, there is nothing to fetch, but one can use it to:
- confirm an endpoint is reachable
- fetch the headers

Reply:
I was going to say "don't you also need a planmodifier for setting `GET` as default"... then I realised this is not a resource :(

Reply:
Fair point. Have added `HEAD`.